README.md

```diff
@@ -4,111 +4,218 @@ An API for validating survey schemas.
+
+Validator runs on two ports, `5001` is the main validator app and `5002` is Ajv validator.
+
+### Validator
+
+Validator runs on `http://localhost:5001/validate` and accepts GET and POST requests.
+
+If you need to change the port you can change the port variable in the Uvicorn settings in api.py:
 ```python
-uvicorn.run("api:app", workers=20, port=5001, reload=False)
+uvicorn.run("api:app", workers=20, port=5001, reload=True)
 ```
-The validator can be called directly in the browser using the "/validate" endpoint and the "url" parameter for the address where the schema is located (eg. GitHub Gist raw json). Only URLs from the ONSDigital organization are accepted.
+The reload flag here will allow the service to restart if you make a change to the code, if you want to run the app locally using multiple server workers you need to set reload to "False".
+
+### Ajv validator
+
+Ajv validator defaults to running on `http://localhost:5002`.
+
+You can override this by setting the `AJV_VALIDATOR_SCHEME`, `AJV_VALIDATOR_HOST`, and `AJV_VALIDATOR_PORT` environment variables.
+The defaults for these are:
+- `AJV_VALIDATOR_SCHEME` = "http"
+- `AJV_VALIDATOR_HOST` = "localhost"
+- `AJV_VALIDATOR_PORT` = "5002"
+
+Alternatively, you can override the entire URL by setting the `AJV_VALIDATOR_URL` environment variable directly.
+
+### Running against a URL
+
+Once Validator is running, it can be called directly in the browser using the "/validate" endpoint and the "url" parameter for the address where the schema is located (e.g. GitHub Gist raw json).
+
+As Validator runs on `localhost:5001` by default, here is an example of a command you can use to validate a schema via a URL:
+```shell
+http://localhost:5001/validate?url=https://raw.githubusercontent.com/ONSdigital/eq-questionnaire-runner/refs/heads/main/schemas/test/en/test_address.json
 ```
-http://localhost:5001/validate?url=...
+
+Only the following URLs and domains are accepted:
+- "https://gist.githubusercontent.com/"
+- "https://raw.githubusercontent.com/"
+- "onsdigital.uk"
+- "localhost"
+
+Also when using a URL from GitHub you can only validate schemas from the ONSdigital organisation and only against repos with the owner "ONSdigital".
+
+### Running against eQ Runner
+
+Also once you have Validator running it can be used to run against eQ Runner (`https://github.com/ONSdigital/eq-questionnaire-runner`).
+If you have eQ Runner spun up you can from within the root of eQ Runner run:
+```shell
+make validate-test-schemas
 ```
+This script will run Validator on the test eQ Runner schemas.
 
+### Running the Ajv (server) version of Validator
+
+Running `make run` will start up the both of services required for validation (Ajv validator and the Validator app itself). However, if you want to start Ajv individually, run:
+```shell
+make start-ajv
+```
 
-## Testing
+This defaults to running on port `5002`, set `AJV_VALIDATOR_PORT` in your .env file if you need to change this.
```

Comment:
We're making references to `.env` files in the readme in a few places now, but there is currently no `.env` file included in this repo. I think we might need a have a short section about how to create & populate this file with var overrides? 

Either that, or maybe have a `.env`file with sensible dev defaults, like we do for the Runner repo? Could be a separate ticket for that though 🤔